backend/manager/modules/vdsbroker/src/main/java/org/ovirt/engine/core/vdsbroker/VdsManager.java

```diff
@@ -1293,11 +1333,40 @@ public void updateV2VJobInfos(List<V2VJobInfo> v2vJobInfos) {
         }
     }
 
+
     private void initAvailableCpus() {
         List<VdsCpuUnit> cpuTopology = cachedVds.getCpuTopology();
-        if (cpuTopology == null) {
+        if (cpuTopology == null || cachedVds.getVdsmCpusAffinity() == null) {
             return;
         }
         this.cpuTopology = cpuTopology;
+        this.cpuTopology.stream().filter(cpu -> cpu.getCpu() == Integer.parseInt(cachedVds.getVdsmCpusAffinity())).forEach(cpu -> cpu.setVmId(Guid.SYSTEM));
+    }
+
+    public List<VdsCpuUnit> getCpuTopology() {
+        List<VdsCpuUnit> clone = new ArrayList<>();
+        cpuTopology.forEach(cpu -> clone.add(cpu.clone()));
+        return clone;
+    }
+
+    public void unpinVmCpus(VM vm) {
+        String pinning = vm.getCpuPinningPolicy() == CpuPinningPolicy.MANUAL ? vm.getCpuPinning() : vm.getCurrentCpuPinning();
+        if (pinning != null && !pinning.isBlank()) {
+            synchronized (this) {
+                cpuTopology.stream().filter(cpu -> CpuPinningHelper.getAllPinnedPCpus(pinning).contains(cpu.getCpu())).forEach(cpu -> {
+                    cpu.setVmId(null);
+                    cpu.setCpuPinningPolicy(CpuPinningPolicy.NONE);
+                });
+            }
+        }
+        // Release the pinned CPUs anyway if there are any on the same VM ID when not having them on the VM itself.
+        synchronized (this) {
```

Comment:
so why having the block above if we're going to change all the vms that are associated with this vm anyway here?

Reply:
I answered on that before, I think it's the best "safe" approach. Of course one of them should be enough.

Reply:
Removed this second block..